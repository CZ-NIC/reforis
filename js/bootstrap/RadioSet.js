/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import {LABEL_SIZE, FIELD_SIZE} from './constants';

export default function RadioSet({name, id, label, choices, value, onChange, ...props}) {
    const radios = choices.map((choice, key) => {
        return <Radio
            key={key}
            name={name}
            label={choice.label}
            value={choice.value}
            onChange={onChange}
            checked={choice.value === value}
            {...props}
        />;
    });

    return <div id={id} className='form-group row'>
        <div className={'form-label col-sm-' + LABEL_SIZE}>
            <label className='form-label' htmlFor={id}>{label}</label>
        </div>
        <div className={'col-sm-' + FIELD_SIZE}>{radios}</div>
    </div>;
}

function Radio({name, label, value, onChange, checked = false,...props}) {
    let id = name + value;
    return <div className='form-check form-check-inline'>
        <input
            className='form-check-input'
            type='radio'
            name={name}
            id={id}
            value={value}
            onChange={onChange}
            checked={checked}
            {...props}
        />
        <label className='form-check-label' htmlFor={id}>{label}</label>
    </div>;
}


