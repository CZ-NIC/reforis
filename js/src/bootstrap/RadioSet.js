/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import propTypes from 'prop-types';
import {useUID} from 'react-uid';

import {LABEL_SIZE, FIELD_SIZE} from './constants';


RadioSet.propTypes = {
    name: propTypes.string.isRequired,
    label: propTypes.string.isRequired,
    choices: propTypes.arrayOf(propTypes.shape({
        label: propTypes.string.isRequired,
        value: propTypes.oneOfType([propTypes.string, propTypes.number]).isRequired,
    })).isRequired,
    value: propTypes.string,
    helpText: propTypes.string,
};

export default function RadioSet({name, label, choices, value, helpText, ...props}) {
    const uid = useUID();
    const radios = choices.map((choice, key) => {
        return <Radio
            id={`${name}-${key}`}
            key={key}
            name={name}
            label={choice.label}
            value={choice.value}
            checked={choice.value === value}
            {...props}
        />;
    });

    return <div id={uid} className='form-group row'>
        <div className={'form-label col-sm-' + LABEL_SIZE}>
            <label className='form-label' htmlFor={uid}>{label}</label>
        </div>
        <div className={'col-sm-' + FIELD_SIZE}>
            {radios}
            <small className="form-text text-muted">{helpText}</small>
        </div>
    </div>;
}

Radio.propTypes = {
    label: propTypes.string.isRequired,
    id: propTypes.string.isRequired,
};

function Radio({label, id, ...props}) {
    return <div className='form-check form-check-inline'>
        <input
            id={id}
            className='form-check-input'
            type='radio'
            {...props}
        />
        <label className='form-check-label' htmlFor={id}>{label}</label>
    </div>;
}


