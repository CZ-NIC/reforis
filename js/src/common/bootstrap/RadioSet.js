/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import PropTypes from 'prop-types';
import {useUID} from 'react-uid/dist/es5/index';

import {formFieldsSize} from './constants';


RadioSet.propTypes = {
    /** Name attribute of the input HTML tag.*/
    name: PropTypes.string.isRequired,
    /** RadioSet label .*/
    label: PropTypes.string,
    /** Choices .*/
    choices: PropTypes.arrayOf(PropTypes.shape({
        /** Choice lable .*/
        label: PropTypes.string.isRequired,
        /** Choice value .*/
        value: PropTypes.oneOfType([PropTypes.string, PropTypes.number]).isRequired,
    })).isRequired,
    /** Initial value .*/
    value: PropTypes.string,
    /** Help text message .*/
    helpText: PropTypes.string,
};

export default function RadioSet({name, label, choices, value, helpText, ...props}) {
    const uid = useUID();
    const radios = choices.map((choice, key) =>
        <Radio
            id={`${name}-${key}`}
            key={key}
            name={name}
            label={choice.label}
            value={choice.value}
            helpText={choice.helpText}
            checked={choice.value === value}

            {...props}
        />
    );

    return <div className={`form-group ${formFieldsSize}`} style={{marginBottom: '1rem'}}>
        {label ?
            <label className='col-12' htmlFor={uid} style={{paddingLeft: '0'}}>
                {label}
            </label>
            : null}
        {radios}
        {helpText ? <small className="form-text text-muted">{helpText}</small> : null}
    </div>;
}

Radio.propTypes = {
    label: PropTypes.string.isRequired,
    id: PropTypes.string.isRequired,
    helpText: PropTypes.string
};

function Radio({label, id, helpText, ...props}) {
    return <>
        <div className='custom-control custom-radio custom-control-inline'>
            <input
                id={id}
                className='custom-control-input'
                type='radio'

                {...props}
            />
            <label className='custom-control-label' htmlFor={id}>{label}</label>
        </div>
        {helpText ? <small className="form-text text-muted">{helpText}</small> : null}
    </>
}
