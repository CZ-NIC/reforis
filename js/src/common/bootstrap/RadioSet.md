Set of radio Bootstrap component input with label and predefined sizes and structure for using in foris forms.

All additional `props` are passed to the `<input type="number">` HTML component.

```js
import {useState} from 'react';
const CHOICES=[
    {value:'one',label:'1'},
    {value:'two',label:'2'},
    {value:'three',label:'3'},
];
const [value, setValue] = useState(CHOICES[0].value);

<>
    {/*Yeah, it gets event, not value!*/}
    <RadioSet
        value={value}
        name='some-radio'
        choices={CHOICES}
        onChange={event=>setValue(event.target.value)}
    />
    <p>Selected value: {value}</p>
</>
```
