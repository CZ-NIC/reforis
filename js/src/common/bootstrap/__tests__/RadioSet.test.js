/*
 * Copyright (C) 2019 CZ.NIC z.s.p.o. (http://www.nic.cz/)
 *
 * This is free software, licensed under the GNU General Public License v3.
 * See /LICENSE for more information.
 */

import React from 'react';
import {render} from 'react-testing-library'

import RadioSet from '../RadioSet';

const TEST_CHOICES = [
    {label: 'label', value: 'value'},
    {label: 'another label', value: 'another value'},
    {label: 'another one  label', value: 'another on value'}
];

describe('<RadioSet/>', () => {
    it('Render radio set', () => {
        const {container} = render(
            <RadioSet
                name={'test_name'}
                label='Radios set label'
                value='value'
                choices={TEST_CHOICES}
                helpText={'Some help text'}
                onChange={() => {
                }}
            />
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});