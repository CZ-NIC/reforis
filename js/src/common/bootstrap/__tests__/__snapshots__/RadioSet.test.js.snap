// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<RadioSet/> Render radio set 1`] = `
<div
  class="form-group col-sm-12 offset-lg-1 col-lg-10"
  style="margin-bottom: 1rem;"
>
  <label
    class="col-12"
    for="1"
    style="padding-left: 0px;"
  >
    Radios set label
  </label>
  <div
    class="custom-control custom-radio custom-control-inline"
  >
    <input
      checked=""
      class="custom-control-input"
      id="test_name-0"
      name="test_name"
      type="radio"
      value="value"
    />
    <label
      class="custom-control-label"
      for="test_name-0"
    >
      label
    </label>
  </div>
  <div
    class="custom-control custom-radio custom-control-inline"
  >
    <input
      class="custom-control-input"
      id="test_name-1"
      name="test_name"
      type="radio"
      value="another value"
    />
    <label
      class="custom-control-label"
      for="test_name-1"
    >
      another label
    </label>
  </div>
  <div
    class="custom-control custom-radio custom-control-inline"
  >
    <input
      class="custom-control-input"
      id="test_name-2"
      name="test_name"
      type="radio"
      value="another on value"
    />
    <label
      class="custom-control-label"
      for="test_name-2"
    >
      another one  label
    </label>
  </div>
  <small
    class="form-text text-muted"
  >
    Some help text
  </small>
</div>
`;
